gensim/models/indexer.py

```diff
@@ -0,0 +1,246 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2013 Radim Rehurek <me@radimrehurek.com>
+# Licensed under the GNU LGPL v2.1 - http://www.gnu.org/licenses/lgpl.html
+
+
+"""
+Using 3rd party k-NN libraries to improve performance for method DocvecsArray.most_similar().
+More information can be found at: https://github.com/piskvorky/gensim/wiki/Ideas-&-Feature-proposals#integrate-a-fast-k-nn-library
+"""
+
+from doc2vec import Doc2Vec
+from word2vec import Word2Vec
+from annoy import AnnoyIndex
```

Comment:
We already have a subpackage for indexing, called `gensim.similarities` -- I think the indexers belong there.

It's not a problem to import `annoy`, but the import cannot happen automatically, on `import gensim` or `import gensim.similarities` (=in `__init__.py`). That would fail for people who don't have Annoy installed, and we definitely don't want to make Annoy a mandatory dependency for gensim.

But we can let users import a module that needs annoy manually: `from gensim.similarities.annoy_index import AnnoyIndex`, that should be fine.